src/transformers/generation/flax_utils.py

```diff
@@ -613,10 +626,17 @@ def greedy_search_body_fn(state):
             logits = model_outputs.logits[:, -1]
 
             # apply min_length, ...
-            logits = logits_processor(state.sequences, logits, state.cur_len)
-
-            next_token = jnp.argmax(logits, axis=-1)
+            next_tokens_scores = logits_processor(state.sequences, logits, state.cur_len)
 
+            next_token = jnp.argmax(next_tokens_scores, axis=-1)
+            if output_scores:
```

Comment:
Thanks for pushing this so that we can see the design played out. I'm not sure this is necessarily a good idea to have all of this control flow and building tensors on the fly in the generate method - did you try running the model with `output_scores=True` and checking we can still jit the generate method? Note that in general it's good to add any relevant tests to confirm that the new behaviour is as expected. I'm a bit worried that this complicates the code and might become tricky for jit'ing Flax generate (which has been written to be a fast generate method)